docs/aqua-book/language/modularity.md

```diff
@@ -0,0 +1,157 @@
+
+```aqua
+-- `aqua` expression may only be on the very first line of the file
+aqua AquaFile
+```
+
+Module name can contain dots, e.g. `aqua Aqua.File`.
+
+This name is used when the module is imported with `use`, see [Importing other modules with `use`](#with-use).
+
+## Specifying what is declared by the module with `declares`
+
+The `aqua AquaFile` expression may optionally include a `declares` section. This section enumerates the elements that the module will make available for **other modules that import it**. If the `declares` section is omitted, the module does not declare anything for other modules to use.
+
+```aqua
+-- This module declares `CONST_NAME`, `ServiceName`, `MyType` and `fn`
+aqua AquaFile declares CONST_NAME, ServiceName, MyType, fn
+
+const CONST_NAME = "something"
+
+service ServiceName:
+  do_something()
+  
+data MyType:
+  result: i32  
+
+func fn() -> string:
+  <- CONST_NAME
+```
+
+To declare everything contained in the file, use `declares *`:
+
+```aqua
+-- This module declares `CONST_NAME`, `ServiceName`, `MyType` and `fn`
+aqua AquaFile declares *
+
+const CONST_NAME = "something"
+
+service ServiceName:
+  do_something()
+  
+data MyType:
+  result: i32  
+
+func fn() -> string:
+  <- CONST_NAME
+```
+
+Note that symbols declared with `declares` are not exported to the host language. To export symbols to the host language, use [`export`](#exporting-to-the-host-language-with-export).
+
+## Importing other modules
+
+Aqua modules can import other modules to use their [declarations](#specifying-what-is-declared-by-the-module-with-declares). There are two ways to import a module: with [`import`](#with-import) and [`use`](#with-use).
+
+### With `import`
+
+The main way to import a module is via `import`. Everything declared in the imported module comes into the current namespace directly.
+
+```aqua
+aqua AquaFile declares foo
+
+-- builtin.aqua declares `Op`
+import "@fluencelabs/aqua-lib/builtin.aqua"
+
+func foo():
+  Op.noop()
+```
+
+It is possible to cherry-pick and rename imports using `import ... from`:
+
+```aqua
+aqua AquaFile declares foo
+
+-- builtin.aqua declares `Op`
+import Op as Noop from "@fluencelabs/aqua-lib/builtin"
+
+func foo():
+  Noop.noop()
+```
+
+### With `use`
+
+The `use` expression makes it possible to import a module as a named scope. The name of the scope is taken from [`aqua` header](#giving-a-name-to-an-aqua-module-with-aqua) of the imported module. Everything declared in the imported module is available in the current namespace as a member of the scope.
+
+```aqua
+aqua AquaFile declares foo
+
+-- builtin.aqua's header is `aqua BuiltIn declares *`
+use "@fluencelabs/aqua-lib/builtin.aqua"
+
+func foo():
+  BuiltIn.Op.noop()
+```
+
+It is possible to rename the imported module with `use ... as ...` expression:
+
+```aqua
+use "@fluencelabs/aqua-lib/builtin" as Renamed
```

Comment:
As far as I remember, you can also do `use x as y from "file" as F`, right? It helps to make a module as an ability to conform a given ability type.